src/request.c

```diff
@@ -83,22 +83,27 @@ void acceptGroupReport(uint32_t src, uint32_t group) {
         my_log(LOG_DEBUG, 0, "Should insert group %s (from: %s) to route table. Vif Ix : %d",
             inetFmt(group,s1), inetFmt(src,s2), sourceVif->index);
 
-        // If we don't have a whitelist we insertRoute and done
+        // If we don't have a black- and whitelist we insertRoute and done
         if(sourceVif->allowedgroups == NULL)
         {
             insertRoute(group, sourceVif->index, src);
             return;
         }
+
         // Check if this Request is legit on this interface
-        struct SubnetList *sn;
+        struct SubnetList   *match = NULL;
+        struct SubnetList   *sn;
+
         for(sn = sourceVif->allowedgroups; sn != NULL; sn = sn->next)
             if((group & sn->subnet_mask) == sn->subnet_addr)
-            {
-                // The membership report was OK... Insert it into the route table..
-                insertRoute(group, sourceVif->index, src);
-                return;
+                match = sn;
+
+        if(match == NULL || (match != NULL && match->allow == 1)) {
+            // The membership report was OK... Insert it into the route table..
+            insertRoute(group, sourceVif->index, src);
```

Comment:
This has an issue. If config file contains only `whitelist` keywords then everything is allowed, even those ranges which are not listed in `whitelist` keyword.

Reply:
Yes, again you are correct. Guess I did not test enough. Issue seems solved now.